### PR TITLE
Prepare documentation for Vue 3 wrapper

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR adds documentation for Vue 3 wrapper.

_[skip changelog]_

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #7545 
2. #8727

### Affected project(s):
- [x] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Comment by budnix** [comment]
@kirszenbaum The PR is yours :) I prepared the docs for the Vue3 wrapper from the technical side. For me everything looks good, all demos are fixed and works with the latest Vue framework. 

As soon as you are done please do not merge the docs to the base branch (`feature/issue-7545`) or `develop` it will deploy all of it to the production. AFAIK the wrapper will be released as RC without official docs yet.

**Comment by kirszenbaum** [comment]
@budnix - thanks! I committed my changes. Separately, I need to rewrite each "Integrate with..." section, but it shouldn't hold up the release for now

**Review by kirszenbaum — APPROVED** [review]

**Comment by budnix** [comment]
> Just one question about the Vue 2 [Using the hot-column component](https://handsontable.com/docs/vue-hot-column/#declaring-a-custom-renderer-as-a-component) section: I understand it shouldn't be present in the Vue 3 docs?

It should! The Vue 3 allows using `hot-column` components. There is only a lack of support for declarative renderers and editors. That is why the [page](https://github.com/handsontable/handsontable/pull/8991/files#diff-6357bd30eb94cf0d78c66a030ab86caf77255864aa31cfc99dbb7b63da644af5) is not extensive as its counterpart Vue 2. I cut the advanced example that uses the mentioned features.

**Comment by kirszenbaum** [comment]
@budnix - sorry, that's what I meant (everything after the `Array of objects` section). All is good then

**Comment by kirszenbaum** [comment]
@budnix - I'm adding the link you mentioned [here](https://github.com/handsontable/handsontable/pull/8727#issuecomment-986682568): ed69c21

Btw, when and how are we planning to merge #8991 ?

**Comment by budnix** [comment]
@kirszenbaum What about the recently introduced changes to how the frameworks are titled? The same should be applied to the Vue 3?

image


> Btw, when and how are we planning to merge #8991 ?

I don't know what is the next step. If we want to release the wrapper as RC the docs should not be merged (thus deployed) .... or merged but with a notation (as a warning tooltip) that it's Release Candidate? 🤔  Alternatively, release with upcoming HoT 11.1.0. cc @warpech

**Comment by kirszenbaum** [comment]
> @kirszenbaum What about the recently introduced changes to how the frameworks are titled? The same should be applied to the Vue 3?

@budnix #9003 will change the wrapper titles once again. #8991 is already aligned with #9003 (I'll update it in case there are further changes)

**Comment by warpech** [comment]
Thanks for nearly having these docs finished! 

@budnix I need to point out that this page needs information about building the Vue 3 wrapper: https://github.com/handsontable/handsontable/blob/docs/issue-7545/docs/next/guides/building-and-testing/building.md. Especially considering the fact that we want to invite early adopters to try the wrapper before the release.

File: docs/next/guides/integrate-with-vue3/vue-modules.md
Comment (kirszenbaum) [inline docs/next/guides/integrate-with-vue3/vue-modules.md:16]:
```suggestion
## Using modules with Vue 3
```

**Review by kirszenbaum — APPROVED** [review]

**Comment by budnix** [comment]
@kirszenbaum Can I ask you to re-review the PR?

**Review by kirszenbaum — DISMISSED** [review]
@budnix - just one question: do we want the Vue 3 tile larger than the other frameworks?

**Comment by budnix** [comment]
> @budnix - just one question: do we want the Vue 3 tile larger than the other frameworks?

Fixed within https://github.com/handsontable/handsontable/pull/8991/commits/d709dc9259a465f59b0b2bef90b0ba8d348a6877.

**Review by kirszenbaum — APPROVED** [review]

**Pull request merged**
